imgix/urlbuilder.py

```diff
@@ -88,3 +110,59 @@ def create_url(self, path, params={}):
             params=params)
 
         return str(url_obj)
+
+    def create_srcset(self, path, params={}):
+        """
+        Create srcset attribute value with the supplied path and
+        `params` parameters dict.
+        Will generate a fixed-width DPR srcset if a width OR height and aspect
+        ratio are passed in as parameters. Otherwise will generate a srcset
+        with width-descriptor pairs.
+
+        Parameters
+        ----------
+        path : str
+        params : dict
+            Dictionary specifying URL parameters. Non-imgix parameters are
+            added to the URL unprocessed. For a complete list of imgix
+            supported parameters, visit https://docs.imgix.com/apis/url .
+            (default {})
+
+        Returns
+        -------
+        str
+            srcset attribute value
+        """
+
+        width = params['w'] if 'w' in params else None
+        height = params['h'] if 'h' in params else None
+        aspect_ratio = params['ar'] if 'ar' in params else None
+
+        if (width or (height and aspect_ratio)):
+            return self.__build_srcset_DPR(path, params)
+        else:
+            return self.__build_srcset_pairs(path, params)
+
+    def __build_srcset_pairs(self, path, params={}):
+        srcset = ''
+        widths = target_widths()
+
+        for i in range(len(widths)):
+            current_width = widths[i]
+            current_params = params.copy()
+            current_params['w'] = current_width
+            srcset += self.create_url(path, current_params) \
```

Comment:
assuming this is python 3, you can use f string formatting and this will look nicer and also run slightly faster

Reply:
I had that in earlier commits but removed it as I think we're technically still supporting python2 until end of the year